### PR TITLE
test: Add an E2E test for deploying an it using secret key filtering

### LINKED ISSUE #3563: Add E2E test for deploying an integration using configmap/secret key filtering
Follow-up on #3543 #3544.

Could you please add test runs with key filtering scenarios after the following ones for both configmap and secret? https://github.com/apache/camel-k/blob/main/e2e/global/common/config/config_test.go#L84
https://github.com/apache/camel-k/blob/main/e2e/global/common/config/config_test.go#L143

### PR DESCRIPTION
**Pull request by essobedo**
fixes #3563

## Motivation

An E2E was missing in https://github.com/apache/camel-k/pull/3544 and had to be added in a different PR

## Modifications

* Define a secret with multi values
* Add a new ECE test based on https://camel.apache.org/camel-k/next/configuration/runtime-config.html#runtime-config-keys  

**Release Note**
```release-note
NONE
```

### CODE REVIEW

**Review by tadayosi — APPROVED** [review]
thx!

File: e2e/global/common/config/config_test.go [hunk 1]
```diff
@@ -167,6 +173,15 @@ func TestRunConfigExamples(t *testing.T) {
 			Eventually(AutogeneratedConfigmapsCount(ns), TestTimeoutShort).Should(Equal(0))
 		})
 
+		t.Run("Secret with filtered key", func(t *testing.T) {
+			Expect(KamelRunWithID(operatorID, ns, "./files/config-secret-key-route.groovy", "--config", "secret:my-sec-multi/my-secret-key-2").Execute()).To(Succeed())
+			Eventually(IntegrationPodPhase(ns, "config-secret-key-route"), TestTimeoutLong).Should(Equal(corev1.PodRunning))
+			Eventually(IntegrationConditionStatus(ns, "config-secret-key-route", v1.IntegrationConditionReady), TestTimeoutShort).Should(Equal(corev1.ConditionTrue))
+			Eventually(IntegrationLogs(ns, "config-secret-key-route"), TestTimeoutShort).ShouldNot(ContainSubstring(cmDataMulti["my-secret-key"]))
+			Eventually(IntegrationLogs(ns, "config-secret-key-route"), TestTimeoutShort).Should(ContainSubstring(cmDataMulti["my-secret-key-2"]))
```
Comment (tadayosi) [inline e2e/global/common/config/config_test.go:181]: These should really use `secDataMulti` not `cmDataMulti`, that's why the test is failing. Please fix it.

**Review by tadayosi — CHANGES_REQUESTED** [review]
The test is failing.

File: e2e/global/common/config/config_test.go (hunk 1 above)
Comment (essobedo) [reply e2e/global/common/config/config_test.go]: My bad, fixed

**Comment by essobedo** [comment]
The added test passes https://github.com/apache/camel-k/runs/8085498798?check_suite_focus=true#step:4:4527

The failing tests are known:

* FAIL: TestKameletBinding/test_error_handler/throw_error_test
* FAIL: TestBindingWithImage/run_with_initial_image
* FAIL: TestErrorHandlerTrait/Run_errored_integration_with_error_handler
* FAIL: TestNativeIntegrations/automatic_rollout_deployment_from_fast-jar_to_native_kit

**Review by tadayosi — APPROVED** [review]

**Comment by tadayosi** [comment]
Thanks!

**Pull request merged**